dags/sql/create-function-generic_find_gaps.sql

```diff
@@ -0,0 +1,109 @@
+--noqa: disable=TMP, PRS
+CREATE OR REPLACE FUNCTION public.generic_find_gaps(
+    start_date date,
+    end_date date,
+    id_col text,
+    dt_col text,
+    sch_name text,
+    tbl_name text,
+    gap_threshold interval,
+    default_bin interval,
+    id_col_dtype anyelement DEFAULT NULL::int
+)
+RETURNS TABLE (
+    sensor_id_col anyelement, gap_start timestamp, gap_end timestamp
+)
+LANGUAGE plpgsql
+COST 100
+VOLATILE
+
+AS $BODY$
+DECLARE
+   resulttype regtype := pg_typeof(id_col_dtype);
+BEGIN
+    RETURN QUERY EXECUTE FORMAT($$
+        WITH raw AS (
+            SELECT DISTINCT
+                CAST(%I AS %s) AS sensor_id_col, --id_col, resulttype
```

Comment:
I feel like we dont need to re-cast the column, as it should already has that column type. is this here so we can declare something?

Reply:
I don't fully understand this, but when I remove this CAST, `generic_find_gaps` still works, but `summarize_gaps_data_check` gives this error: 

> ERROR:  Returned type integer does not match expected type regtype in column 1.structure of query does not match function result type 
> 
> ERROR:  structure of query does not match function result type
> SQL state: 42804
> Detail: Returned type integer does not match expected type regtype in column 1.
>